frontend/src/app/testQueue/QueueItem.test.tsx

```diff
@@ -0,0 +1,123 @@
+import { MockedProvider } from "@apollo/client/testing";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { act } from "react-dom/test-utils";
+
+import QueueItem, { EDIT_QUEUE_ITEM } from "./QueueItem";
+
+describe("QueueItem", () => {
+  it("correctly renders the test queue", () => {
+    const { container, getByTestId } = render(
+      <MockedProvider mocks={[]}>
+        <QueueItem
+          internalId={testProps.internalId}
+          patient={testProps.patient}
+          askOnEntry={testProps.askOnEntry}
+          selectedDeviceId={testProps.selectedDeviceId}
+          selectedDeviceTestLength={testProps.selectedDeviceTestLength}
+          selectedTestResult={testProps.selectedTestResult}
+          devices={testProps.devices}
+          defaultDevice={testProps.defaultDevice}
+          refetchQueue={testProps.refetchQueue}
+          facilityId={testProps.facilityId}
+          dateTestedProp={testProps.dateTestedProp}
+          patientLinkId={testProps.patientLinkId}
+        ></QueueItem>
+      </MockedProvider>
+    );
+    expect(screen.getByText("Potter, Harry James")).toBeInTheDocument();
+    expect(getByTestId("timer")).toHaveTextContent("10:00");
+    expect(container).toMatchSnapshot();
+  });
+
+  it("updates the timer when a device is changed", async () => {
+    const { getByTestId, getByLabelText } = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <QueueItem
+          internalId={testProps.internalId}
+          patient={testProps.patient}
+          askOnEntry={testProps.askOnEntry}
+          selectedDeviceId={testProps.selectedDeviceId}
+          selectedDeviceTestLength={testProps.selectedDeviceTestLength}
+          selectedTestResult={testProps.selectedTestResult}
+          devices={testProps.devices}
+          defaultDevice={testProps.defaultDevice}
+          refetchQueue={testProps.refetchQueue}
+          facilityId={testProps.facilityId}
+          dateTestedProp={testProps.dateTestedProp}
+          patientLinkId={testProps.patientLinkId}
+        ></QueueItem>
+      </MockedProvider>
+    );
+    await act(async () => {
+      fireEvent.change(getByLabelText("Device", { exact: false }), {
+        target: { value: "lumira" },
+      });
+      await new Promise((resolve) => setTimeout(resolve, 0));
+    });
+    expect(getByTestId("timer")).toHaveTextContent("15:00");
```

Comment:
Great test!

Reply:
Thanks! It took me forever to figure this out so I appreciate you saying that 🤣